### PR TITLE
[20.09] Restore the ability to pull containers from admin menu

### PR DESCRIPTION
**Pull request by mvdbeek**
This used to work in the old dependency menu because it went through
https://github.com/mvdbeek/galaxy/blob/dev/lib/galaxy/tool_util/deps/__init__.py#L263

### CODE REVIEW

File: lib/galaxy/tool_util/deps/container_resolvers/mulled.py [hunk 1]
```diff
@@ -408,16 +408,19 @@ def resolve(self, enabled_container_types, tool_info, install=False, session=Non
...
             )
             destination_for_container_type = kwds.get('destination_for_container_type')
-            if install and destination_for_container_type and not self.cached_container_description(
+            if install and not self.cached_container_description(
                     targets,
                     namespace=self.namespace,
                     hash_func=self.hash_func,
                     resolution_cache=resolution_cache,
             ):
+                destination = {}
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/deps/container_resolvers/mulled.py]:
```suggestion
                destination_info = {}
```

File: lib/galaxy/tool_util/deps/container_resolvers/mulled.py [hunk 2]
```diff
@@ -408,16 +408,19 @@ def resolve(self, enabled_container_types, tool_info, install=False, session=Non
...
+                if destination_for_container_type:
+                    destination = destination_for_container_type(self.container_type)
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/deps/container_resolvers/mulled.py]:
```suggestion
                    destination_info = destination_for_container_type(self.container_type)
```

File: lib/galaxy/tool_util/deps/container_resolvers/mulled.py [hunk 3]
```diff
@@ -408,16 +408,19 @@ def resolve(self, enabled_container_types, tool_info, install=False, session=Non
...
                 container = CONTAINER_CLASSES[self.container_type](container_description.identifier,
                                                                    self.app_info,
                                                                    tool_info,
-                                                                   destination_for_container_type(self.container_type),
+                                                                   destination,
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/deps/container_resolvers/mulled.py]:
```suggestion
                                                                   destination_info,
```

**Review by nsoranzo — COMMENTED** [review]
Also `destination_for_container_type` should probably be renamed to `destination_info_for_container_type` here and at https://github.com/galaxyproject/galaxy/blob/release_20.09/lib/galaxy/tool_util/deps/__init__.py#L263

**Comment by mvdbeek** [comment]
Let's do that in dev ?

File: lib/galaxy/tool_util/deps/container_resolvers/mulled.py [hunk 4]
```diff
@@ -408,16 +408,19 @@ def resolve(self, enabled_container_types, tool_info, install=False, session=Non
                 shell=self.shell,
             )
             destination_for_container_type = kwds.get('destination_for_container_type')
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/deps/container_resolvers/mulled.py]:
```suggestion
```

File: lib/galaxy/tool_util/deps/container_resolvers/mulled.py [hunk 5]
```diff
@@ -408,16 +408,19 @@ def resolve(self, enabled_container_types, tool_info, install=False, session=Non
...
-            if install and destination_for_container_type and not self.cached_container_description(
+            if install and not self.cached_container_description(
                     targets,
                     namespace=self.namespace,
                     hash_func=self.hash_func,
                     resolution_cache=resolution_cache,
             ):
+                destination_info = {}
+                if destination_for_container_type:
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/deps/container_resolvers/mulled.py:418]:
```suggestion
                destination_for_container_type = kwds.get('destination_for_container_type')
                if destination_for_container_type:
```

**Comment by nsoranzo** [comment]
> Let's do that in dev ?

Sure

**Review by nsoranzo — APPROVED** [review]

File: lib/galaxy/tool_util/deps/container_resolvers/mulled.py [hunk 6]
```diff
@@ -407,17 +407,20 @@ def resolve(self, enabled_container_types, tool_info, install=False, session=Non
                 type=self.container_type,
                 shell=self.shell,
             )
-            destination_for_container_type = kwds.get('destination_for_container_type')
-            if install and destination_for_container_type and not self.cached_container_description(
```
Comment (dannon) [inline lib/galaxy/tool_util/deps/container_resolvers/mulled.py:411]: It's this bit right?  It dropped the bypass when a destination is trusted to understand a container type implicitly?

Comment (mvdbeek) [reply lib/galaxy/tool_util/deps/container_resolvers/mulled.py:411]: Are you looking at a bug or weird behavior ? If you wanted to fetch a container previously you had to pass in `destination_for_container_type `, which you don't have available when not preparing a job, for instance when pulling an image from the admin panel. Is this causing excessive pulls, and if so, what does the container resolution config look like ?

Comment (mvdbeek) [reply lib/galaxy/tool_util/deps/container_resolvers/mulled.py:411]: Ah, just saw the issue now.

**Pull request merged**